### PR TITLE
vsmile_cart.xml: Add a new UK game cart dump and use hex for ROM sizes

### PR DESCRIPTION
**Pull request by clawgrip**
New working software list additions
---------------------------------------
Disney's Winnie the Pooh - The Honey Hunt (UK) [apex2504]

### CODE REVIEW

File: hash/vsmile_cart.xml [hunk 1]
```diff
@@ -737,9 +737,9 @@ V.Smile Smartbook Smartidges (need a Smartbook touch tablet connected to a regul
 			<feature name="cart_type" value="lilac" />
 			<feature name="u1" value="" /> <!-- EPOXY BLOB ROM --> <!-- 2 globs, uses /CS2 for 2nd glob -->
 			<feature name="u2" value="" /> <!-- EPOXY BLOB ROM -->
-			<dataarea name="rom" size="0x0800000">
-				<rom name="alphaprkr2.u1" size="0x0400000" crc="4e635c94" sha1="96792d607a94f8fb36dbfff8c68f720c68c742a8" offset="0x0000000" />
-				<rom name="alphaprkr2.u2" size="0x0200000" crc="2feee80a" sha1="d1ca700602c3b2bce2a8920f017370cb0ef322de" offset="0x0400000" />
+			<dataarea name="rom" size="0x600000">
+				<rom name="alphaprkr2.u1" size="0x400000" crc="4e635c94" sha1="96792d607a94f8fb36dbfff8c68f720c68c742a8" offset="0x000000" />
+				<rom name="alphaprkr2.u2" size="0x200000" crc="2feee80a" sha1="d1ca700602c3b2bce2a8920f017370cb0ef322de" offset="0x400000" />
```
Comment (cuavas) [inline hash/vsmile_cart.xml:742]: Are you sure this won’t cause a problem?  The slot code may assume a minimum size for the ROM region.  Also, if you’re going to mess with this stuff, please put the offsets next to the sizes so they’re easier to see.

File: hash/vsmile_cart.xml [hunk 2]
```diff
@@ -788,7 +788,7 @@ V.Smile Smartbook Smartidges (need a Smartbook touch tablet connected to a regul
 			<feature name="cart_type" value="lilac" />
 			<feature name="u1" value="U1" />
 			<feature name="u2" value="U2" />
-			<dataarea name="rom" size="8388608">
+			<dataarea name="rom" size="0x600000">
```
Comment (cuavas) [inline hash/vsmile_cart.xml:791]: As with the other one where you changed the size, have you checked the code to see whether it makes assumptions about the region size?

File: hash/vsmile_cart.xml [hunk 3]
```diff
@@ -837,9 +837,9 @@ V.Smile Smartbook Smartidges (need a Smartbook touch tablet connected to a regul
 			<feature name="cart_type" value="lilac" />
 			<feature name="u1" value="" /> <!-- EPOXY BLOB ROM --> <!-- 2 globs, uses /CS2 for 2nd glob -->
 			<feature name="u2" value="" /> <!-- EPOXY BLOB ROM -->
-			<dataarea name="rom" size="0x0800000">
-				<rom name="092003.u1" size="0x0400000" crc="402681b7" sha1="5ff1ba9a016d10e09408b5638aaf759cac2cc97a" offset="0x0000000" /> <!-- U1 is the same as U1 on 80-092000(US) -->
-				<rom name="092003.u2" size="0x0200000" crc="60cfb1b4" sha1="f37ad4dd774c26abb998f616ab7ad0adce83359d" offset="0x0400000" />
+			<dataarea name="rom" size="0x600000">
+				<rom name="092003.u1" size="0x400000" crc="402681b7" sha1="5ff1ba9a016d10e09408b5638aaf759cac2cc97a" offset="0x000000" /> <!-- U1 is the same as U1 on 80-092000(US) -->
+				<rom name="092003.u2" size="0x200000" crc="60cfb1b4" sha1="f37ad4dd774c26abb998f616ab7ad0adce83359d" offset="0x400000" />
```
Comment (cuavas) [inline hash/vsmile_cart.xml:842]: Once again, have you checked the slot code?  And please make put the offsets next to the sizes.

File: hash/vsmile_cart.xml [hunk 4]
```diff
@@ -2413,9 +2413,9 @@ V.Smile Smartbook Smartidges (need a Smartbook touch tablet connected to a regul
 			<feature name="pcb" value="92060" />
 			<feature name="u1" value="" /> <!-- EPOXY BLOB ROM --> <!-- 2 globs, uses /CS2 for 2nd glob -->
 			<feature name="u2" value="" /> <!-- EPOXY BLOB ROM -->
-			<dataarea name="rom" size="0x0800000">
-				<rom name="vsmile92100.blob1.bin" size="0x0400000" crc="8a2317fb" sha1="44d397ada00f1fedb48622e716189f4a129ff5a7" offset="0x0000000" />
-				<rom name="vsmile92100.blob2.bin" size="0x0200000" crc="dded069c" sha1="29f1f464f9f5c8fa06410d54538b984f3fc52f01" offset="0x0400000" />
+			<dataarea name="rom" size="0x600000">
+				<rom name="vsmile92100.blob1.bin" size="0x400000" crc="8a2317fb" sha1="44d397ada00f1fedb48622e716189f4a129ff5a7" offset="0x000000" />
+				<rom name="vsmile92100.blob2.bin" size="0x200000" crc="dded069c" sha1="29f1f464f9f5c8fa06410d54538b984f3fc52f01" offset="0x400000" />
```
Comment (cuavas) [inline hash/vsmile_cart.xml:2418]: Same applies here.

File: hash/vsmile_cart.xml [hunk 5]
```diff
@@ -4016,9 +4016,9 @@ V.Smile Smartbook Smartidges (need a Smartbook touch tablet connected to a regul
 			<feature name="slot" value="vsmile_rom" />
 			<feature name="u1" value="" /> <!-- EPOXY BLOB ROM --> <!-- 2 globs, uses /CS2 for 2nd glob -->
 			<feature name="u2" value="" /> <!-- EPOXY BLOB ROM -->
-			<dataarea name="rom" size="0x0800000">
-				<rom name="092143.u1" size="0x0400000" crc="987d99d6" sha1="78901c0029d0c4df7f151c7654c86bc43147e4cd" offset="0x0000000" />
-				<rom name="092143.u2" size="0x0200000" crc="3eae10c6" sha1="1124233305e3b28753007c57ef356a2351cbd117" offset="0x0400000" />
+			<dataarea name="rom" size="0x600000">
+				<rom name="092143.u1" size="0x400000" crc="987d99d6" sha1="78901c0029d0c4df7f151c7654c86bc43147e4cd" offset="0x000000" />
+				<rom name="092143.u2" size="0x200000" crc="3eae10c6" sha1="1124233305e3b28753007c57ef356a2351cbd117" offset="0x400000" />
```
Comment (cuavas) [inline hash/vsmile_cart.xml:4021]: Same applies here.

File: hash/vsmile_cart.xml [hunk 6]
```diff
@@ -4766,9 +4766,9 @@ V.Smile Smartbook Smartidges (need a Smartbook touch tablet connected to a regul
 			<feature name="pcb" value="92060" />
 			<feature name="u1" value="" /> <!-- EPOXY BLOB ROM --> <!-- 2 globs, uses /CS2 for 2nd glob -->
 			<feature name="u2" value="" /> <!-- EPOXY BLOB ROM -->
-			<dataarea name="rom" size="0x0800000">
-				<rom name="vsmile92060.blob1.bin" size="0x0400000" crc="c34fd808" sha1="2800b89068e3c09a66223392e11982dd2193244b" offset="0x0000000" />
-				<rom name="vsmile92060.blob2.bin" size="0x0200000" crc="b7261f74" sha1="40bc8d9284a2f307023502ec67f8fc8e974966ea" offset="0x0400000" />
+			<dataarea name="rom" size="0x600000">
+				<rom name="vsmile92060.blob1.bin" size="0x400000" crc="c34fd808" sha1="2800b89068e3c09a66223392e11982dd2193244b" offset="0x000000" />
+				<rom name="vsmile92060.blob2.bin" size="0x200000" crc="b7261f74" sha1="40bc8d9284a2f307023502ec67f8fc8e974966ea" offset="0x400000" />
```
Comment (cuavas) [inline hash/vsmile_cart.xml:4771]: Same applies here.

File: hash/vsmile_cart.xml [hunk 7]
```diff
@@ -4798,23 +4798,42 @@ V.Smile Smartbook Smartidges (need a Smartbook touch tablet connected to a regul
...
+			<feature name="slot" value="vsmile_rom" />
+			<feature name="pcb" value="706148" />
+			<feature name="pcb_model" value="706148-3" />
+			<feature name="cart_type" value="lilac" />
+			<feature name="u1" value="" />
+			<feature name="u2" value="" />
+			<dataarea name="rom" size="0x0800000">
+				<rom name="vsmile92063.blob1.bin" size="0x0400000" crc="24efb09a" sha1="c67a19edcc521e987fee546f3a0bc4bc0dbcfdb5" offset="0x0000000"/>
+				<rom name="vsmile92063.blob2.bin" size="0x0200000" crc="b7261f74" sha1="40bc8d9284a2f307023502ec67f8fc8e974966ea" offset="0x0400000" />
+			</dataarea>
```
Comment (cuavas) [inline hash/vsmile_cart.xml:4822]: Please put the offsets next to the sizes.

**Review by cuavas — CHANGES_REQUESTED** [review]
I’m pretty sure the V.Smile cartridge code assumes the cartridge ROM region is 8M (0x800000) and will read off the end if the supplied region is too small.  You *definitely* shouldn’t be making smaller cartridge ROM regions, and you should really reload to mirror up to the end of the region or set a fill flag depending on whether the cartridge is fully decoded.

**Comment by clawgrip** [comment]
Thanks!!

**Pull request merged**